### ISSUE TITLE
FindMany returns wrong type after extending prisma client

### ISSUE THREAD

**Issue by niba** [prisma/prisma#19864]
### Bug description

Types inference breaks when you pass query as **any** to the findMany function in extended prisma client;

```ts
  const result = prisma.model.findMany(query as any);
  const id = result[0].id; 
  /*
     result is 
    (GetResult<{MODEL_FIELDS}, unknown> & {})[]
  */

  const extendedPrisma = prisma.$extends({});
  const result = extendedPrisma.model.findMany(query as any); // you need to pass query as any to get invalid types
  const id = result[0].id // ts error
  /*
    result is 
      ({} | (GetResult<{MODELS_FIELDS}, {
          ...;
      }> & {}))[]
  */
```

### How to reproduce

--

### Prisma information

--

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: 18

### Prisma Version

```
4.16.0-dev.78
```

**Comment by Jolg42** [comment]
@niba Thanks for the bug issue!

It's now published in our `patch-dev` version `4.16.1-dev.4`.

Let us know if this version fixes the issues, we're curious to get any feedback 🙌🏼 

Note: do not use in production, wait for the official `4.16.1` patch for that.

**Comment by iigorcunha** [comment]
After the update for 4.16 and 4.16.1, if I don't pass my entity to overwrite the return of findMany. I will see an error because of the GetResult typo from Prisma.

**Comment by millsp** [comment]
@iigorcunha which typo? Could you please show an example of what is your project, ideally please open a new issue. Thanks.

**Comment by iigorcunha** [comment]
Sure @millsp I will open a new issue. thank you.

**Comment by millsp** [comment]
Thanks! Also, you could try `[4.16.2-dev.2](https://www.npmjs.com/package/@prisma/client/v/4.16.2-dev.2)` as we have some staged fixes for the next patch release, maybe they fix your problem?

**Comment by Jolg42** [comment]
Note, the official `4.16.2` patch is now available.

**Comment by GuysmoB** [comment]
Hi,

![image](https://github.com/prisma/prisma/assets/42407006/dd49b5d3-43e5-4249-a4ab-f25f7a5e0318)

I have this weird return type : GetResult< .....Unknow> even if I force the type. 
I bet it wasn't like that few weeks ago. Anyone can explain me this please ?

Thanks for any help :)

**Comment by janpio** [comment]
That was a change that was unintended and will be fixed in 5.1 in a few weeks. See https://github.com/prisma/prisma/issues/20320 for the issue that reported this and was closed via a PR already.

**Comment by GuysmoB** [comment]
Thank you sir !
